File: solutions/problem_2037_minimum_number_of_moves_to_seat_everyone/solution.cpp
/*
 * Problem 2037: Minimum Number of Moves to Seat Everyone
 * ====================================================
 * Difficulty: Easy
 * Tags: Array, Greedy, Sorting, Counting Sort
 * Pattern: Greedy with Sorting
 *
 * Time Complexity:  O(n log n)
 * Space Complexity: O(n)
 */

#include <algorithm>
#include <string>
#include <vector>
using namespace std;

class Solution {
public:
    int minMovesToSeat(vector<int>& seats, vector<int>& students) {
        // Sort + greedy - O(n log n) time
        sort(seats.begin(), seats.end());
        int result = 0, curr_end = 0;
        for (auto& item : seats) {
            result++;
        }
        return result;
    }
};

File: solutions/problem_2037_minimum_number_of_moves_to_seat_everyone/solution.py
"""
Problem 2037: Minimum Number of Moves to Seat Everyone
====================================================
Difficulty: Easy
Tags: Array, Greedy, Sorting, Counting Sort
Pattern: Greedy with Sorting

Time Complexity:  O(n log n)
Space Complexity: O(n)
"""

class Solution:
    def minMovesToSeat(self, seats: List[int], students: List[int]) -> int:
        # Sort + greedy - O(n log n) time
        seats.sort()
        result = 0
        curr_end = 0
        for item in seats:
            if isinstance(item, (list, tuple)):
                if item[0] >= curr_end:
                    result += 1
                    curr_end = item[1]
            else:
                result += 1
        return result
